frontend/test/pytest/test_measurement_primitives.py

```diff
@@ -93,6 +131,23 @@ def f():
     assert jaxpr.eqns[1].outvars[0].aval.shape == (1,)
 
 
+def test_probs_dynamic():
+    """Test that the probs primitive can be captured by jaxpr
+    when using tracers."""
+
+    def f(dim_0):
+        obs = compbasis_p.bind()
+        return probs_p.bind(obs, shots=5, shape=(dim_0,))
+
+    jaxpr = jax.make_jaxpr(f)(1)
+    shots, shape_value = jaxpr.eqns[1].params.values()
+
+    assert jaxpr.eqns[1].primitive == probs_p
+    assert isinstance(shape_value[0], DynamicJaxprTracer)
+    assert shots == 5
```

Comment:
Do you know what changed that means there are no escaped tracer issues compared to your previous test?

Reply:
Note the jaxpr for the example here is different than what JAX would generate, and I'm wondering if that isn't going to lead to issues (like escaped tracers).

Our jaxpr, contains tracer in the jaxpr equation:
```py
{ lambda ; a:i64[]. let
    b:AbstractObs(num_qubits=0,primitive=compbasis) = compbasis 
    c:f64[a] d:i64[a] = counts[
      shape=(Traced<ShapedArray(int64[], weak_type=True)>with<DynamicJaxprTrace(level=1/0)>,)
      shots=5
    ] b
  in (c, d) }
```

JAX's jaxpr for similar example (`jnp.ones((dim_0, 5))`), does not insert the tracer as constant data into the jaxpr equation:
```py
{ lambda ; a:i64[]. let
    b:f64[a,5] = broadcast_in_dim[broadcast_dimensions=() shape=(None, 5)] 1.0 a
  in (b,) }
```
Instead we find `None` in the place where the tracer would be.